snapshot/kaniko_snapshot_unix.go

```diff
@@ -0,0 +1,107 @@
+// +build linux darwin
+
+package snapshot
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/buildpacks/lifecycle"
+
+	kconfig "github.com/GoogleContainerTools/kaniko/pkg/config"
+	ksnap "github.com/GoogleContainerTools/kaniko/pkg/snapshot"
+	kutil "github.com/GoogleContainerTools/kaniko/pkg/util"
+)
+
+type KanikoSnapshotter struct {
+	RootDir                    string
+	DetectFilesystemIgnoreList bool
```

Comment:
What's this for? It seems always to be `true` except for the tests?

Reply:
So that bool triggers `DetectFilesystemIgnoreList()`. The reason it is off for tests is that we don't want it detecting the test runner's ignore list. In a real container, this is safe to run and produces additional ignore list items we don't want in the snapshot.

https://github.com/GoogleContainerTools/kaniko/blob/f20f495051627b988c7449e9b163a4f9f30cc044/pkg/util/fs_util.go#L410